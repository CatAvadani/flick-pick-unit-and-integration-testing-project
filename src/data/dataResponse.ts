export interface TrendingResults {
  page: number;
  results: ResultItem[];
  total_pages: number;
  total_results: number;
}

export interface ResultItem {
  id: number;
  title?: string;
  original_title?: string;
  overview: string;
  poster_path?: string;
  vote_average?: number;
  release_date?: string;
}
