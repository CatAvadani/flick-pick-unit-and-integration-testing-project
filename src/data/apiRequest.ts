import { MediaType, TrendingResults } from './dataResponse';

const API_KEY = import.meta.env.VITE_API_KEY;

export const getTrending = async (
  page: number = 2
): Promise<TrendingResults> => {
  const response = await fetch(
    `https://api.themoviedb.org/3/trending/all/day?language=en-US&api_key=${API_KEY}&page=${page}`
  );
  const data = await response.json();
  return data;
};

export const getSearchResults = async (
  query: string
): Promise<TrendingResults> => {
  console.log('Search query:', query);

  const response = await fetch(
    `https://api.themoviedb.org/3/search/multi?language=en-US&api_key=${API_KEY}&query=${encodeURIComponent(
      query
    )}`
  );
  const data = await response.json();
  return data;
};

export const getMovieDetails = async (id: number, type: MediaType) => {
  const response = await fetch(
    `https://api.themoviedb.org/3/${type}/${id}?language=en-US&api_key=${API_KEY}`
  );
  const data = await response.json();
  return data;
};
