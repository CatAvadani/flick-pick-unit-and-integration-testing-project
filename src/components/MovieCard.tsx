import { Link } from 'react-router-dom';
import { ResultItem } from '../data/dataResponse';
import { truncateText } from '../utils/truncateText';

type MovieCardProps = {
  movie: ResultItem;
};

function MovieCard({ movie }: MovieCardProps) {
  return (
    <Link to={`/movie/${movie.id}`}>
      <div className='w-72 h-full bg-black/5 rounded-sm hover:opacity-70 transition-all'>
        <img
          className='w-full h-48 object-cover rounded-t-sm'
          src={`https://image.tmdb.org/t/p/w500${movie.poster_path}`}
          alt={movie.title}
        />
        <div className='p-4 mt-4 flex flex-col gap-8'>
          <h2 className='text-xl font-bold text-white'>{movie.title}</h2>
          <p className='text-white/50'>
            {truncateText(movie.overview, 10)}{' '}
            <span className=' text-amber-400'>Read more &gt;&gt;</span>
          </p>
        </div>
      </div>
    </Link>
  );
}

export default MovieCard;
