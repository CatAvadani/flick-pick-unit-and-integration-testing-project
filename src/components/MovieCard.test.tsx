import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { describe, expect, it } from 'vitest';
import { ResultItem } from '../data/dataResponse';
import MovieCard from './MovieCard';

describe('MovieCard Component', () => {
  const movie: ResultItem = {
    id: 1,
    title: 'Game of Thrones',
    overview:
      'Seven noble families fight for control of the mythical land of Westeros',
    poster_path: '/poster_title.jpg',
  };

  it('should render movie title', () => {
    render(
      <MemoryRouter>
        <MovieCard movie={movie} />
      </MemoryRouter>
    );
    expect(screen.getByText('Game of Thrones')).toBeInTheDocument();
  });

  it('should render movie poster with correct src', () => {
    render(
      <MemoryRouter>
        <MovieCard movie={movie} />
      </MemoryRouter>
    );
    const img = screen.getByAltText('Game of Thrones') as HTMLImageElement;
    expect(img.src).toBe('https://image.tmdb.org/t/p/w500/poster_title.jpg');
  });

  it('should have a link that navigates to the correct movie details page', () => {
    render(
      <MemoryRouter>
        <MovieCard movie={movie} />
      </MemoryRouter>
    );

    const link = screen.getByRole('link') as HTMLAnchorElement;
    expect(link).toHaveAttribute('href', '/movie/1');
  });
});
