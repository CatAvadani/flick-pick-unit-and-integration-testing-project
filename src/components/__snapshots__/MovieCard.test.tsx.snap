// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MovieCard Component > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="w-72 h-full bg-black/5 rounded-sm hover:opacity-70 transition-all"
  >
    <img
      alt="Game of Thrones"
      class="w-full h-48 object-cover rounded-t-sm"
      src="https://image.tmdb.org/t/p/w500/poster_title.jpg"
    />
    <div
      class="p-4 mt-4 flex flex-col gap-8"
    >
      <h2
        class="text-xl font-bold text-white"
      >
        Game of Thrones
      </h2>
      <p
        class="text-white/50"
      >
        Seven noble families fight for control of the mythical land... 
        <span
          class=" text-amber-400"
        >
          Read more &gt;&gt;
        </span>
         
      </p>
    </div>
  </div>
</DocumentFragment>
`;
